channeldb/payments.go

```diff
@@ -147,7 +147,9 @@ const (
 	// balance to complete the payment.
 	FailureReasonInsufficientBalance FailureReason = 4
 
-	// TODO(halseth): cancel state.
+	// FailureReasonCanceled indicates that the payment was canceled by the
+	// user.
+	FailureReasonCanceled FailureReason = 5
```

Comment:
need to update the `String` method below, also wondering if it's easy to modify one of the itest cases to check this behavior (probably not as we don't have an easy way to cancel context there). Mentioning the test because I think there are other places that also need to be updated, such as `lnrpc.PaymentFailureReason`.

Reply:
Thank you for noticing this, I've adjusted the `String` method and complemented the `lnrpc` part.

For an `itest` we'd probably have to create harness rpc method like `SendPaymentWithContext` or so, then pre-cancel it and try sending the payment, which should result in `lnrpc.FAILURE_REASON_CANCELED`.

Should we do that?

Reply:
I am unsure if we should introduce `lnrpc.Payment_CANCELED` because adding this state would be a deviation from how we handle other payment `FailureReason`s.

In the reference below we define state `lnrpc.Payment_FAILED` as an umbrella state for any `FailureReason`. So adding a further distinction here would complicate the flow. 
https://github.com/lightningnetwork/lnd/blob/8c0d7862c210ea6db5dc90ebfe9c3236962e68d4/channeldb/payment_status.go#L180

I think the proper way to check for a cancel is first to check the payment for `lnrpc.Payment_FAILED`, then check the payment failure reason for `PaymentFailureReason_FAILURE_REASON_INCORRECT_PAYMENT_DETAILS`.

Reply:
yea my bad that code ref is incorrect as we def don't want to add a new status, what I meant is,
```go
	stream := alice.RPC.SendPaymentWithContext(ctx, req)
	ht.AssertPaymentStatusFromStream(stream, lnrpc.Payment_IN_FLIGHT)

	ctx.Cancel()
	payment := ht.AssertPaymentStatusFromStream(stream, lnrpc.Payment_FAILED)

	require.Equal(ht, lnrpc.PaymentFailureReason_FAILURE_REASON_CANCELED, payment.FailureReason)
```

Reply:
If there is nothing from preventing the payment from settling then the payment status after the `ctx.Cancel()` should be `lnrpc.Payment_SUCCEEDED` since the cancellation can't abort the current payment attempt. But the failure reason should be canceled.

If attached an itest.